dashboard/app/models/levels/blockly.rb

```diff
@@ -362,14 +366,14 @@ def localized_authored_hints
     return unless authored_hints
 
     if should_localize?
-      translations = get_localized_property("authored_hints")
+      authored_hints_key = get_localized_property("authored_hints", resolve: false)
 
-      return unless translations.instance_of? Hash
+      return unless authored_hints_key
 
       localized_hints = JSON.parse(authored_hints).map do |hint|
         next if hint['hint_markdown'].nil? || hint['hint_id'].nil?
 
-        translated_text = translations.try(:[], hint['hint_id'].to_sym)
+        translated_text = I18n.t(hint['hint_id'], scope: authored_hints_key, default: nil)
```

Comment:
is there a specific reason why you're using `scope` here rather than 

    I18n.t("#{authored_hints_key}.#{hint['hint_id']}")

?

Reply:
no specific reason, `scope` and string concatenation/interpolation with the separator (dot) are equivalent (micro-optimizations aside). I think I just had a slight preference for `scope` here just because the scope and key strings were both already formed earlier.